test/connectObservable.test.tsx

```diff
@@ -266,6 +266,39 @@ describe("connectObservable", () => {
     )
   })
 
+  it("allows errors to be caught in error boundaries with suspense", async () => {
+    const errStream = new Subject()
+    const [useError] = connectObservable(errStream)
+
+    const ErrorComponent = () => {
+      const value = useError()
+
+      return <>{value}</>
+    }
+
+    const errorCallback = jest.fn()
+    render(
+      <TestErrorBoundary onError={errorCallback}>
+        <Suspense fallback={<div>Loading...</div>}>
+          <ErrorComponent />
+        </Suspense>
+      </TestErrorBoundary>,
+    )
+
+    componentAct(() => {
+      errStream.error("controlled error")
+    })
+
+    await componentAct(async () => {
+      await wait(0)
+    })
```

Comment:
Ewww.

I couldn't find any doc on this, but by experimenting, it seems that when a thrown promise fails in Suspense:
* It creates a new component instance. Yep, the reducer `init` function gets called, and if you have a `setState(() => Math.random())` you'll get another value.
* It retries rendering the component a couple of times.

It looks like the single `componentAct` can't pick this up. Chaining two toghether does fix it.

Do you smell that smelly smell?